k8s/object.bzl

```diff
@@ -172,13 +172,22 @@ def _common_impl(ctx):
 
   kubectl_tool_info = ctx.toolchains["@io_bazel_rules_k8s//toolchains/kubectl:toolchain_type"].kubectlinfo
 
+  kubectl_args = " ".join(ctx.attr.kubectl_args or [])
```

Comment:
Ok, so lets make this simple then, lets start w/o supporting make variable expansion and just avoid needing all this code (we can add it later if we need to).
Based off that assumption, all you need here is this line:
`kubectl_args = " ".join(ctx.attr.kubectl_args or [])`

and everything between lines 176-180 can go away.
and then just do 
`"%{kubectl_args}": kubectl_args,`
in line 190 below

Reply:
> we can add it later if we need to

Is there a reason we don't add it now?  I will update the code as you recommended, just want to learn more :)